src/cattrs/gen.py

```diff
@@ -717,7 +717,7 @@ def make_mapping_structure_fn(
             lines.append("      errors.append(e)")
             lines.append("  if errors:")
             lines.append(
-                f"    raise IterableValidationError('While structuring {cl!r}', errors, __cattr_mapping_cl)"
+                f"    raise IterableValidationError('While structuring ' + {repr(cl)!r}, errors, __cattr_mapping_cl)"
```

Comment:
This might look odd at first glance - the first `repr()` gets a string representation of the class and the second repr (`!r`) turns said string into a string literal